src/ring_theory/subring.lean

```diff
@@ -486,7 +488,68 @@ instance : complete_lattice (subring R) :=
 lemma eq_top_iff' (A : subring R) : A = ⊤ ↔ ∀ x : R, x ∈ A :=
 eq_top_iff.trans ⟨λ h m, h $ mem_top m, λ h m _, h m⟩
 
-/-! # subring closure of a subset -/
+/-! ## Center of a ring -/
+
+variable (R)
+
+/--
+  The center of a ring `R` is the set of elements of `R` which commute with everything
+  under multiplication.
+-/
+def center : subring R :=
+{ carrier := {a | ∀ r, a * r = r * a},
+  one_mem' := by simp,
+  mul_mem' := λ a b (ha : ∀ r, a * r = r * a) (hb : ∀ r, b * r = r * b) s,
+    by rw [mul_assoc, ha, mul_assoc, hb, mul_assoc],
+  zero_mem' := by simp,
+  add_mem' := λ a b (ha : ∀ r, a * r = r * a) (hb : ∀ r, b * r = r * b) s,
+    by rw [add_mul, mul_add, ha, hb],
+  neg_mem' := λ a (ha : ∀ r, a * r = r * a) s,
+    by rw [←neg_mul_eq_mul_neg, ←neg_mul_eq_neg_mul, ha] }
```

Comment:
Does it make sense to define this for `subsemiring` first, and then just populate the single `neg_mem` field from there?

Reply:
I think so yeah, this definition doesn't involve any addition specific structure.

Reply:
It sounds like what you're describing is to define `submonoid.center` first? (which would also be great, but is more work!)

Reply:
Well, considering `subgroup.center` already exists, I'll have to think about how that'll work.

Reply:
I would imagine you can do something like:
```lean
def submonoid.center :=
{ carrier := {a | ∀ r, a * r = r * a},
  one_mem' := by simp,
  mul_mem' := λ a b (ha : ∀ r, a * r = r * a) (hb : ∀ r, b * r = r * b) s,
    by rw [mul_assoc, ha, mul_assoc, hb, mul_assoc],

def subgroup.center :=
{ inv_mem' := sorry,
  .. submonoid.center }

def subsemiring.center :=
{ zero_mem' := sorry,
  add_mem' := sorry,
  .. submonoid.center }
 
def subring.center :=
{ neg_mem' := sorry,
  .. subsemiring.center }
```
(spread over the files of course)